### PR TITLE
Upadtes to framework for running on bare NANOAOD

### PR DESCRIPTION
**Pull request by sroychow**
Test on Z config :- 
```
Running on full dataset
{ "/scratchnvme/wmass/BARENANO/preVFP/Run2016B/merged/tree_2.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016B/merged/tree_1.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016C/merged/tree_1.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016D/merged/tree_2.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016D/merged/tree_1.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016E/merged/tree_1.root", "/scratchnvme/wmass/BARENANO/preVFP/Run2016F/merged/tree_1.root" }
processing  data
Post nano node name:  postnano
this is how I will make variations for this histogram
dimuonMass { "" }
Mu1_eta { "" }
Mu1_pt { "" }
Mu2_eta { "" }
Mu2_pt { "" }
compiling
this is how I will make variations for this histogram
dimuonMass { "" }
dimuonPt { "" }
dimuonY { "" }
MET_pt { "" }
compiling
{ "/scratchnvme/wmass/BARENANO/preVFP/DYJetsToMuMu/merged/tree_2.root", "/scratchnvme/wmass/BARENANO/preVFP/DYJetsToMuMu/merged/tree_1.root" }
DYJetsToMuMu_M50 235996656029.08496
processing  DYJetsToMuMu_M50
Post nano node name:  postnano
this is how I will make variations for this histogram
dimuonMass { "" }
Mu1_eta { "" }
Mu1_pt { "" }
Mu2_eta { "" }
Mu2_pt { "" }
lumiweight { "" }
puWeight { "" }
SF { "" }
compiling
this is how I will make variations for this histogram
dimuonMass { "" }
dimuonPt { "" }
dimuonY { "" }
MET_pt { "" }
lumiweight { "" }
puWeight { "" }
SF { "" }
compiling
data
DYJetsToMuMu_M50
data
data_muons
data_dimuon
471431418 events processed in 3.4 s rate 138384032.6761263
DYJetsToMuMu_M50
DY_muons
DY_dimuon
118919995 events processed in 3.2 s rate 36910093.14709467
all samples processed in 248.5967206954956 s

```

### CODE REVIEW

File: templateMaker/dySequence.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        return p
+    elif systType == 1:
+        p.branch(nodeToStart = 'defs', nodeToEnd = 'defs', modules = [getLumiWeight(xsec=xsec, inputFile = fvec, genEvsbranch = "genEventSumw", targetLumi = 19.3), ROOT.SF_ul(fileSFul, isZ=True)])
+        p.Histogram(columns = ["dimuonMass", "Mu1_eta", "Mu1_pt", "Mu2_eta", "Mu2_pt", "lumiweight", "puWeight"], types = ['float']*7,node='defs',histoname=ROOT.string('DY_muons'),bins = [zmassBins, etaBins, ptBins, etaBins, ptBins], variations = [])
+        p.EventFilter(nodeToStart='defs', nodeToEnd='defs', evfilter="abs(Mu1_eta) < 2.4 && abs(Mu2_eta) < 2.4 && Mu1_pt > 25. && Mu2_pt > 25. && Mu1_pt < 55. && Mu2_pt < 55.", filtername="{:20s}".format("mu pt eta acceptance"))
+        p.Histogram(columns = ["dimuonMass", "dimuonPt", "dimuonY", "MET_pt", "lumiweight", "puWeight"], types = ['float']*6,node='defs',histoname=ROOT.string('DY_dimuon'),bins = [zmassBins,qtBins, etaBins,metBins], variations = [])
+        return p
+    else:
+        p.branch(nodeToStart = 'defs', nodeToEnd = 'defs', modules = [ROOT.lumiWeight(xsec=xsec, sumwclipped=236188699235.12158, targetLumi = 19.3), ROOT.SF_ul(fileSFul, isZ=True)])
```
Comment (emanca) [inline templateMaker/dySequence.py]: please substitute gen sumW hardcoded number with the relevant variable

File: templateMaker/plotZ_simplifiedV2.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
+import os
```
Comment (emanca) [inline templateMaker/plotZ_simplifiedV2.py]: please move this file to Common

File: templateMaker/src/SF_ul.cpp [hunk 3]
```diff
@@ -12,11 +12,37 @@ RNode SF_ul::run(RNode d)
...
+	//apply the iso no trig since trig m
+	//combinedSF *= _iso_notrig->GetBinContent(binIso1);
+	//combinedSF *= _iso_notrig->GetBinContent(binIso2) ;
+	
+	//int binIso1 = _iso->FindBin(eta1, pt1);
+	//combinedSF *= _iso->GetBinContent(binIso1);
+	//int binIso2 = _iso->FindBin(eta2, pt2);
+	//combinedSF *= _iso->GetBinContent(binIso2);
+	/*
         if (_prefCharge == charge1 && istrigMatched1 > 0)
```
Comment (emanca) [inline templateMaker/src/SF_ul.cpp]: please clean unneeded lines

File: templateMaker/src/getZmassV2.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#include "getZmassV2.hpp"
```
Comment (emanca) [inline templateMaker/src/getZmassV2.cpp]: time to rename this class! what about Zselection?
ps Defines here can be made using RVec syntax
check /scratchnvme/emanca/wproperties-analysis/templateMaker/src/Zselection.cpp

File: templateMaker/src/recoDefinitions.cpp [hunk 5]
```diff
@@ -13,6 +13,7 @@ RNode recoDefinitions::run(RNode d)
                   .Define("Mu1_pt", getFromIdx, {"Muon_pt", "Idx_mu1"})
                   .Define("Mu1_sip3d", getFromIdx, {"Muon_sip3d", "Idx_mu1"})
                   .Define("Mu1_dxy", getFromIdx, {"Muon_dxy", "Idx_mu1"})
+                  .Define("Mu1_hasTriggerMatch", getIntFromIdx, {"Muon_hasTriggerMatch", "Idx_mu1"})
```
Comment (emanca) [inline templateMaker/src/recoDefinitions.cpp:16]: please remove this line since it will break backward compatibility with old sample

Comment (emanca) [reply templateMaker/src/recoDefinitions.cpp:16]: Idx_mu1 is outdated!

File: templateMaker/runZ_simplifiedV2.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
+import os
```
Comment (emanca) [inline templateMaker/runZ_simplifiedV2.py]: please remove this file

**Review by emanca — CHANGES_REQUESTED** [review]

File: templateMaker/src/recoDefinitions.cpp (hunk 5 above)
Comment (sroychow) [reply templateMaker/src/recoDefinitions.cpp:16]: This will be done in a separate PR when the W Sequence is defined.

**Review by emanca — APPROVED** [review]

**Pull request merged**